ui/apps/platform/src/Containers/Collections/CollectionForm.tsx

```diff
@@ -207,6 +219,38 @@ function CollectionForm({
         setDeleteId(null);
     }
 
+    function onSaveCollection(collection: Collection) {
+        if (action.type === 'view') {
+            // Logically should not happen, but just in case
+            return;
+        }
+
+        const saveServiceCall =
+            action.type === 'edit'
+                ? (payload) => updateCollection(action.collectionId, payload)
+                : (payload) => createCollection(payload);
+
+        const requestPayload = generateRequest(collection);
+        const { request } = saveServiceCall(requestPayload);
+
+        request
+            .then(() => {
+                history.push({ pathname: `${collectionsBasePath}` });
```

Comment:
As a follow up, you might compare to policies which `goBack` after save. For a specific example, there was some consensus among the team that when people have the task to create several collections, it might be convenient and seem intuitive to go back to table page which has **Create collection** button.

We can brainstorm pro and con on a call, if you like.

Reply:
Sequence of titles in browser history is another way to compare pro and con of push versus pop.

Reply:
Sure we can discuss this next week. I compared the behavior of policies and integrations when deciding on this vs `goBack` and went with push since the way the form works seems a little more similar to integrations. In both collections and integrations, the individual view pages are more of a read-only version of the form, while policies has its own dedicated UI for viewing a single policy. I felt like the `goBack` functionality was a little unclear when going from an editable collection form to a read-only collection form with the changes applied.